### PR TITLE
legacy: webuser removal

### PR DESCRIPTION
**Pull request by hachreak**
* INCOMPATIBLE Removes legacy webuser module. (addresses #3233)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: invenio/legacy/webpage.py [hunk 1]
```diff
@@ -98,10 +151,9 @@ def page(title, body, navtrail="", description="", keywords="",
        output: the final cds page with header, footer, etc.
     """
 
-    _ = gettext_set_language(language)
     if req is not None:
         if uid is None:
-            uid = getUid(req)
+            uid = current_user.get_id(req)
```
Comment (jirikuncar) [inline invenio/legacy/webpage.py]: `current_user.get_id()` takes no arguments

File: invenio/testsuite/test_restful.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def put(self):
 
         # Create a user
         self.user = User(
-            email='info@invenio-software.org', nickname='tester'
+            email='test-info@invenio-software.org', nickname='tester'
```
Comment (jirikuncar) [inline invenio/testsuite/test_restful.py]: `info+test@invenio-software.org` or `test@example.com`?

File: invenio/ext/logging/wrappers.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ def get_pretty_wide_client_info(req):
     """Return (in a pretty way) all the available information about the current
     user/client"""
     if req:
-        from invenio.legacy.webuser import collect_user_info
+        from invenio.modules.accounts.helpers import collect_user_info
         user_info = collect_user_info(req)
```
Comment (jirikuncar) [inline invenio/ext/logging/wrappers.py]: maybe you can consider using:
```diff
+ from flask_login import current_user as user_info
- from invenio.legacy.webuser import collect_user_info
- user_info = collect_user_info(req)
```

File: invenio/ext/login/legacy_user.py [hunk 4]
```diff
@@ -59,6 +67,20 @@
 }
 
 
+def isUserAdmin(user_info):
```
Comment (jirikuncar) [inline invenio/ext/login/legacy_user.py]: What about passing `user` object directly to both `isUserAdmin` and `isUserSuperAdmin`?

Comment (hachreak) [inline invenio/ext/login/legacy_user.py]: They are called only inside `legacy_user.py` itself and from legacy `bibdocfile`.
I think we can consider it as legacy and remove after all legacy code will be removed.

File: invenio/ext/login/legacy_user.py [hunk 5]
```diff
@@ -59,6 +67,20 @@
...
+    """Return True if the user has some admin rights; False otherwise."""
+    user = User.query.get(user_info['uid'])
+    return user and user.has_admin_role
+
+def isUserSuperAdmin(user_info):
+    """Return True if the user is superadmin; False otherwise."""
+    user = User.query.get(user_info['uid'])
+    if user and user.has_super_admin_role:
+        return True
+    return acc_firerole_check_user(
```
Comment (hachreak) [inline invenio/ext/login/legacy_user.py]: @jirikuncar do you think we need to move this part directly inside the [has_super_admin_role](https://github.com/inveniosoftware/invenio/blob/master/invenio/modules/access/models.py#L255) property?

Comment (jirikuncar) [inline invenio/ext/login/legacy_user.py]: @hachreak no, because `has_super_admin_role` says only if a user has super admin role attached.

**Comment by jirikuncar** [comment]
@hachreak please rebase and update dependent packages. Thanks

**Comment by hachreak** [comment]
@jirikuncar  rebased! :)

**Pull request merged**